experimental/docker-build-with-squash.md

```diff
@@ -0,0 +1,49 @@
+# Docker build --squash
+
+`--squash` is a experimental CLI argument to docker build that allows your built images to be squashed to their parent, which does not destory any images or layers, and preserves the build cache.
+
+Once the build is complete, docker creates a new image loading the diffs from each layer into a single new layer and references all the parent's layers.
+
+## The following is an example of docker build with `--squash` argument
+
+```Dockerfile
+
+FROM busybox
+RUN echo hello > /hello
+RUN echo world >> /hello
+RUN touch remove_me /remove_me
+ENV HELLO world
+RUN rm /remove_me
+
+```
+An image named `test` is built with `--squash`.
+
+```bash
+$ docker build -t test squash .
```

Comment:
yes, `--squash` is a flag, so the command should be `docker build --squash -t test .`